treeherder/intermittents_commenter/commenter.py

```diff
@@ -324,17 +325,47 @@ def get_bug_stats(self, startday, endday):
         bug_map = self.build_bug_map(bugs, option_collection_map)
         return bug_map, bug_ids
 
+    def fetch_test_variants(self):
+        mozilla_central_url = "https://hg.mozilla.org/mozilla-central"
+        variant_file_url = f"{mozilla_central_url}/raw-file/tip/taskcluster/kinds/test/variants.yml"
```

Comment:
If a variant gets removed and the failures of last week processed, it will be missing in the former but present in the latter. This is a rare edge case and we can leave with it.